### PR TITLE
limit function

### PR DESCRIPTION
**Pull request by nkint**
I'm re-writing some p5.js example with gl-vec2 and `limit` function is missing in this repository, so here it is! ✨

### CODE REVIEW

**Comment by rreusser** [comment]
Looks great! Two nitpicks.

1. Could you inline the math? From what I can see, none of the other functions require other functions. When it comes down to it, this is (to the best of my knowledge) because the stackgl gl-matrix modules are designed to be at least as fast as reasonably possible, which isn't always great but is at least measurably better if you inline math instead of abstracting out all the possible function calls. In my experience, this has actually made a fairly significant difference to the extent that I often even remove dependencies on the stackgl gl-matrix modules and inline this code inside loops when performance really matters.

2. Could you add this function to the readme?

Thanks!

**Comment by nkint** [comment]
Ok, @rreusser thanks for the nitpicks tell me if there is something else! : )

File: limit.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+module.exports = limit;
+
+var squaredLength = require("./squaredLength");
+var scale = require("./scale");
+var copy = require("./copy");
```
Comment (rreusser) [inline limit.js]: Could you remove these three requires?

File: limit.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Limit the magnitude of this vector to the value used for the `max`
+ * parameter.
+ *
+ * @method limit
+ * @param  the vector to limit
+ * @param  max the maximum magnitude for the vector
+ * @returns out
+ */
+function limit(out, a, max) {
+  var mSq = squaredLength(a);
```
Comment (rreusser) [inline limit.js]: And inline this (i.e. a[0] * a[0] + a[1] + a[1] + a[2] * a[2])

**Comment by rreusser** [comment]
Thanks, @nkint, just a couple remaining tweaks, I think!

**Comment by nkint** [comment]
@rreusser ok thanks, I forget to remove unused import 😅
also fixed the doc

**Comment by nkint** [comment]
@rreusser any news on this?

**Comment by rreusser** [comment]
Oops, I missed the previous notification, @nkint. Thanks so much for the module and for making those changes!

**Comment by rreusser** [comment]
I've published v1.3.0 with this addition. Thanks!

**Pull request merged**